### PR TITLE
#168781691 approve/reject request

### PR DESCRIPTION
**Pull request by alainmateso**
#### What does this PR do?
Give a manager the ability to approve or reject a request

#### Description of Task to be completed?
- adds the possibility for a manager to view requests awaiting their decision
- adds the possibility for requesters to view all requests they have submitted
- adds the possibility for a manager to approve or reject requests

#### How should this be manually tested?
- Clone the repository from [here](https://github.com/andela/caret-bn-backend.git)
- Browse to the repo directory
- navigate to ft-approve-reject-request-168781691 branch
- Run `npm install`
- run `sequelize db:migrate`
- run `sequelize db:seed:all`
- run `npm run dev`

- To view requests

* If you are logged in as a manager
Send a GET request to http://localhost:port/api/v1/requests/manager
* If you are logged in as a requester
Send a GET request to http://localhost:port/api/v1/requests

- To approve a request

Log in as a manager
Send a PATCH request to http://localhost:port/api/v1/requests/approve/requestId

- To reject a request

Log in as a manager
Send a PATCH request to http://localhost:port/api/v1/requests/reject/requestId

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168781691
](https://www.pivotaltracker.com/story/show/168781691)

#### Screenshots (if appropriate)
Screen Shot 2019-10-23 at 10 07 05
Screen Shot 2019-10-23 at 20 09 40
Screen Shot 2019-10-23 at 20 09 45

#### Questions:
N/A

### CODE REVIEW

File: src/database/models/requeststatus.js [hunk 1]
```diff
@@ -2,5 +2,13 @@ module.exports = (sequelize, DataTypes) => {
   const requestStatus = sequelize.define('requestStatus', {
     name: DataTypes.STRING
   });
+  // requestStatus.associate = function (models) {
```
Comment (sabin18) [inline src/database/models/requeststatus.js]: remove these comments.

**Review by sabin18 — CHANGES_REQUESTED** [review]
good job @alainmateso ,please update your PR with correct routes, you forgot to add ID.When manager get response he still get that request is pending, it require to click twice.So work on that .

**Review by Cheza-Dzabala — APPROVED** [review]
Checked. I approve.

**Review by sabin18 — APPROVED** [review]
good job @alainmateso

File: src/controllers/requestController.js [hunk 2]
```diff
@@ -1,26 +1,82 @@
...
+import responseHelper from '../utils/responseHelper';
+import responseUtil from '../utils/responseUtil';
 import userServices from '../services/userServices';
 import Utilities from '../utils/index';
 
+const {
+  APPROVED, REJECTED, SUCCESSFULLY_RETRIEVED_REQUESTS
+} = strings.requests;
+const { NO_REQUESTS, ASSIGNED_REQUESTS } = text.user.requests;
+
```
Comment (nakiwuge) [inline src/controllers/requestController.js:14]: Remove this line

File: src/routes/api/requests.js [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
+const {
+  viewMyRequests, approveRequest, rejectRequest, viewManagerRequests
+} = requestController;
 
-router.get('/', validateToken, (req, res) => requestController.viewRequests(req, res));
+const { checkManagerRole } = checkRole;
+
+router.get('/', validateToken, viewMyRequests);
+router.get('/manager', validateToken, checkManagerRole, viewManagerRequests);
+router.patch('/approve/:id', validateToken, checkManagerRole, checkId, approveRequest);
```
Comment (nakiwuge) [inline src/routes/api/requests.js]: Change this to `manager/approve/:id`

File: src/routes/api/requests.js [hunk 4]
```diff
@@ -1,9 +1,19 @@
...
+router.patch('/reject/:id', validateToken, checkManagerRole, checkId, rejectRequest);
```
Comment (nakiwuge) [inline src/routes/api/requests.js]: Change this to `manager/reject/:id`

File: src/controllers/requestController.js [hunk 5]
```diff
@@ -1,26 +1,82 @@
...
 
-    return Utilities.responseHelper(
-      res,
-      Utilities.stringsHelper.user.requests.SUCCESSFULLY_RETRIEVED_REQUESTS,
-      requests,
-      200
-    );
+    const requestToReject = await models.requests.findOne({
+      where: { id }, attributes: ['id'], include: [{ model: models.requestStatus, as: 'status', attributes: ['id', 'name'] }, { model: models.users, attributes: ['id', 'username', 'email'], where: { lineManager: userManager } }]
+    });
```
Comment (nakiwuge) [inline src/controllers/requestController.js]: Create a helper function for this query

**Review by nakiwuge — CHANGES_REQUESTED** [review]
great job @alainmateso,  I have left comments kindly check them out.

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**